cmd/populator/artifactory.go

```diff
@@ -39,20 +38,11 @@ func GetPomFromURL(url string) gopom.Project {
 // GetCoreVersionFromPom get liquibase core version string from POM object
 func GetCoreVersionFromPom(pom gopom.Project) string {
 	var version string
-	for _, dep := range *pom.Dependencies {
-		if *dep.ArtifactID == "liquibase-core" {
-			if strings.Contains(*dep.Version, "${") {
-				v := strings.TrimPrefix(*dep.Version, "${")
-				v = strings.TrimSuffix(v, "}")
-				for k, prop := range pom.Properties.Entries {
-					if k == v {
-						version = prop
-					}
-				}
-			} else {
-				version = *dep.Version
-			}
+	for k, prop := range pom.Properties.Entries {
+		if k == "liquibase.version" {
```

Comment:
This is not an accurate fix. The dependency is on liquibase-core and not an static value of `liquibase.version`. This would not correctly work on extensions that had a version in liquibase-core that's not the liquibase.version property.

Reply:
The liquibase-parent-pom sets de liquibase-core version [as follows:](https://github.com/liquibase/liquibase-parent-pom/blob/9fd035a76c33a96cdd9d941977ad3d7119a6badf/pom.xml#L236)

```
        <dependency>
            <groupId>org.liquibase</groupId>
            <artifactId>liquibase-core</artifactId>
            <version>${liquibase.version}</version>
            <scope>provided</scope>
        </dependency>
```

and extensions have as property:

`<liquibase.version>4.24.0</liquibase.version>`

i.e:

- https://github.com/liquibase/liquibase-bigquery/blob/95e8fdf15a030ee685e412689b3a57161894a9e7/pom.xml#L22C9-L22C54
- https://github.com/liquibase/aws-secrets-manager-extension/blob/67e316cd57c655c06435ae600914612c3d03de64/pom.xml#L33
- https://github.com/liquibase/liquibase-commercial-mongodb/blob/330ca3cd8c03497b69b0303c67bff97b6d941c3f/pom.xml#L24
- https://github.com/liquibase/liquibase-neo4j/blob/0a489adc85b486aebe89564ce56477a2d8c5140c/pom.xml#L48
- https://github.com/liquibase/liquibase-hanadb/blob/798c4d613c8b0c331e38ed91054fd803d88bcfc6/pom.xml#L51
- https://github.com/liquibase/liquibase-commercial-dynamodb/blob/d129391c86ae2a92ed3e929f7e74743aeeb27c6c/pom.xml#L24
- https://github.com/liquibase/liquibase-hibernate/blob/0c51165a6cdc412e399d59dc815842da2e42bc57/pom.xml#L56
- https://github.com/liquibase/liquibase-singlestore/blob/a9d49917c0cd6db0811b85bce7b79ec39869795e/pom.xml#L49
- https://github.com/liquibase/liquibase-redshift/blob/b4f8c9b9dc9788c802ca6543dacf1845c03b621b/pom.xml#L21
- https://github.com/liquibase/liquibase-nochangeloglock/blob/8b88e671f40ee29c43364bb3808b3fde8241be41/pom.xml#L22
- https://github.com/liquibase/liquibase-dbmarlin/blob/2d16817f4c982facefb5fed2843811ff2e5a991f/pom.xml#L40
- https://github.com/liquibase/liquibase-databricks/blob/e01094ccdaa84480ad9fd5bfeebf6f66329542e7/pom.xml#L30

No liquibase-core dependency is defined at the extension level, just the property to be able to be changed, or in this case to be able to be retrieved by LPM when building the package file.

Also, extensions which are not migrated to use the parent pom are setting the liquibase-core version using the same property:

- https://github.com/liquibase/liquibase-percona/blob/f088c320eab19f47c6a8dadff6886b3d46cb086b/pom.xml#L77

There is another case where the property is different like https://github.com/liquibase/liquibase-couchbase/blob/95014cd1a4a61ce3f2833f5e0cfbb8043d30a419/pom.xml#L85

Finally, If there is some extension like this one https://github.com/liquibase/liquibase-oracle/blob/1d9c791d5389fc98df9a6eebbb394a83d6e00bce/pom.xml#L59 which is not using a property to set the version, then it must be changed. We need a standard.

I took that property name when creating the parent pom because it was used in that way.

Migrating all extensions to use the parent-pom would "fix" the small number of cases where extensions do not use liquibase.version or cases where they use "hardcoded" values in the dependency.

Reply:
I think the better solution would be to conditionally check if liquibase-core is defined in the extension pom, traverse and use that version, if not and parent pom is present, get the parent pom, traverse to liquibase core and get the liquibase version from there. It doesn't make assumptions about where to find liquibase core version and follows a pattern of checking if the parent pom is overridden first.

Reply:
This solution looks good.